### DISCUSSION TITLE
[response-cache] Caching for multi-root queries

### DISCUSSION THREAD

**Discussion by Tobbe** [dotansimha/envelop#1370] [Q&A] [answered]
Hi!

I have (a Redwood Cell with) this query
```graphql
query StoreProductQuery($sku: String!, $lang: String) {
  categories(lang: $lang) {
    id
    name
    parent
  }
  product(sku: $sku, lang: $lang) {
    sku
    name
    short_desc
    long_desc
    regular_price
    unavailable
    images
  }
}
```

And this cache config
```js
{
  ttl: 0,
  ttlPerSchemaCoordinate: {
    'Query.allProducts': ONE_HOUR,
    'Query.featuredProducts': ONE_DAY,
    'Query.topSellers': ONE_DAY,
    'Query.rootProducts': ONE_DAY,
    'Query.categories': ONE_DAY,
  },
}
```

With this config the whole query is cached for a day.

I also tested this
```js
{
  ttl: 0,
  ttlPerSchemaCoordinate: {
    'Query.allProducts': ONE_HOUR,
    'Query.featuredProducts': ONE_DAY,
    'Query.topSellers': ONE_DAY,
    'Query.rootProducts': ONE_DAY,
    'Query.categories': ONE_DAY,
    'Query.product': 0,
  },
}
```
With this I get no caching at all.

Is there any way to use cached data for the categories, but fresh data for products?

EDIT:
I realize it makes sense the way it works. It's caching the full response. If one part of the response is invalidated, the whole thing is invalidated.
Sill curious about possible solutions here though, if there are any 🙂  One obvious thing to do would be to split this up in two separate queries. But Redwood's Cells don't support that.

**Comment by n1ru4l** [comment, marked as answer]
Hey @Tobbe

Sorry for the late reply. Right now we don't have the intention of changing the response cache to to modular/normalized caching (similar to GraphQL frontend clients). We are happy to see other people trying to innovate in that space and support those!
